src/vulkan/Context.zig

```diff
@@ -72,13 +74,124 @@ pub fn init(
     };
 
     const instance_handle = try self.vkb.createInstance(&instance_create_info, null);
-    self.vki = try InstanceDispatch.load(instance_handle, self.vkb.dispatch.vkGetInstanceProcAddr);
-    self.instance = Instance.init(instance_handle, &self.vki);
+    const vki = try InstanceDispatch.load(instance_handle, self.vkb.dispatch.vkGetInstanceProcAddr);
+    self.instance = Instance.init(instance_handle, &vki);
+
+    var surface: vk.SurfaceKHR = undefined;
+    if (glfw.createWindowSurface(instance_handle, window, null, &surface) != 0)
```

Comment:
Pass a function pointer instead so we don't have to know about GLFW in this module

Reply:
This is okay here imo as we need multiple defines from glfw (`Window`, `createWindowSurface`) but surface should be a field of `Context` instead of a local. Surface also needs to be destroyed on error in this scope & on deinit of context, validation layers complain about that one